targets/TARGET_ARM_SSG/TARGET_MUSCA_A1/TARGET_MUSCA_A1_S/flash_api.c

```diff
@@ -0,0 +1,132 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "device.h"
+#include "flash_layout.h"
+#include "flash_api.h"
+
+#if DEVICE_FLASH
+
+#define FLASH_DEV          MT25QL_DEV_NS
+
+int32_t flash_init(flash_t *obj)
```

Comment:
this function can be called multiple times without calling de-init, can the driver handle that? Could we add a init flag? Same questions for de-init as well.

Reply:
Good question ,we will need to update flash HAL to add defined/undefined behavior. 

Current driver has a note "Should be called once per lifetime of the object." so this HAL does not do any check, a caller should do.